### PR TITLE
hub/analytics: fix bad DNS

### PR DESCRIPTION
**Pull request by haraldschilly**
# Description
* if you do no set any DNS, this falls back to cocalc.com as a default (was an oversight!)
*  and it also takes care if you enter something that is not "ok" ... tweaking the types with questionsmarks makes this situation clear to TS

# Testing Steps
hub starts, I  did also delete my entry via `delete from server_settings where name = 'dns';`

# Relevant Issues
none, but the stacktrace is

```
TypeError: Cannot read property 'domain' of null
    at /cocalc/src/smc-hub/analytics.ts:269:49
    at Layer.handle [as handle_request] (/cocalc/src/smc-hub/node_modules/express/lib/router/layer.js:95:5)
    at next (/cocalc/src/smc-hub/node_modules/express/lib/router/route.js:137:13)
    at cors (/cocalc/src/smc-hub/node_modules/cors/lib/index.js:188:7)
    at /cocalc/src/smc-hub/node_modules/cors/lib/index.js:224:17
    at origin (/cocalc/src/smc-hub/analytics.ts:241:37)
    at /cocalc/src/smc-hub/node_modules/cors/lib/index.js:219:13
    at optionsCallback (/cocalc/src/smc-hub/node_modules/cors/lib/index.js:199:9)
    at corsMiddleware (/cocalc/src/smc-hub/node_modules/cors/lib/index.js:204:7)
    at Layer.handle [as handle_request] (/cocalc/src/smc-hub/node_modules/express/lib/router/layer.js:95:5)
    at next (/cocalc/src/smc-hub/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/cocalc/src/smc-hub/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/cocalc/src/smc-hub/node_modules/express/lib/router/layer.js:95:5)
    at /cocalc/src/smc-hub/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/cocalc/src/smc-hub/node_modules/express/lib/router/index.js:335:12)
    at next (/cocalc/src/smc-hub/node_modules/express/lib/router/index.js:275:10)
    at /cocalc/src/smc-hub/hub_http_server.coffee:134:14
[...]
```

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- [ ] Run eslint on new and edited files
- [ ] All new code is actually used.
- [ ] Non-obvious code has some sort of comments.

Front end:
- [ ] Restart at least one project and check its `~/.smc/local_hub/local_hub.log`
- [ ] Completely restart Webpack with `./w` in `/src`
- [ ] Completely restart the hub by killing and restarting `./start_hub.py` in `/src/dev/project`
- [ ] Screenshots if relevant.

### CODE REVIEW

File: src/smc-util/db-schema/site-defaults.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@
 // Default settings to customize a given site, typically a private install of CoCalc.
 
 const { is_valid_email_address } = require("smc-util/misc");
+const { DNS } = require("../theme");
```
Comment (williamstein) [inline src/smc-util/db-schema/site-defaults.ts:9]: Change this to be an import (and same for above).  If you use imports then typescript provides potentially useful information via type inference.  If you use require, it disables all of that.  Never ever ever use require unless you have no choice (e.g., coffeescript code).

**Comment by haraldschilly** [comment]
I updated this parser to version 3, which brings significant changes. The code looks and works better now, but I didn't test all cases yet.

**Comment by williamstein** [comment]
Changing to "needs work" to be clear to me that this is waiting on:

> "but I didn't test all cases yet."

**Comment by williamstein** [comment]
I'm going to just merge this.

**Pull request merged**